eox_nelp/signals/receivers.py

```diff
@@ -386,11 +386,15 @@ def pearson_vue_course_completion_handler(instance, **kwargs):  # pylint: disabl
     the pearson vue real_time_import_task , that basically runs the RTI pipeline
     on the completion logic.
     This sends only the user_id and course_id kwargs.
+    user_id -> instance.user_id
+    course_key -> instance.context_key
 
     Arguments:
         instance<Blockcompletion>: Instance of BlockCompletion model.
     """
-    if not getattr(settings, "PEARSON_RTI_ACTIVATE_COMPLETION_GATE", False):
+    if not getattr(settings, "PEARSON_RTI_ACTIVATE_COMPLETION_GATE", False) or not getattr(
+        settings, "PEARSON_RTI_COURSES_DATA", {}
```

Comment:
please consider to update the setting name and type

Reply:
The question is: is someone using the past integration? The idea was to keep compatibility with both integrations.

Anyway why fo you say the type? to change the dict to a list or something like that? I also like the dict approach:
``` json
{
"PEARSON_RTI_COURSES_DATA": {
"course-v1:one": true,
"course-v1:two": true,
"course-v1:three": false

...
}
}
```

Reply:
>The question is: is someone using the past integration? The idea was to keep compatibility with both integrations.

If the setting name has been changed, it doesn’t mean the previous implementation is unavailable. It might require an extra configuration step, which could seem redundant, but it remains compatible.

> Anyway why fo you say the type? to change the dict to a list or something like that? I also like the dict approach:

You mentioned compatibility, but the dictionary approach is indeed completely new; the current implementation does not support that structure. However, if you prefer the dictionary over a simple list, that’s fine. Just update the setting name and adjust the logic to check for the value instead of the key.

Reply:
Actually the v0 version is also compatible looking the test:
https://github.com/eduNEXT/eox-nelp/blob/7785b9bbc78aeabc5ffefd60ef62eb485dd91f77/eox_nelp/signals/tests/test_receivers.py#L703-L710

Is like 
``` json
{
"PEARSON_RTI_COURSES_DATA": {
  "course-v1:one": { 
      "exam_authorization_count": 3, 
      "exam_series_code": "OTT" 
  },
  "course-v1:two": { 
      "exam_authorization_count": 23, 
      "exam_series_code": "OTT" 
  },
  "course-v1:three": { 
      "exam_authorization_count": 13, 
      "exam_series_code": "OTT" 
  }

...
}
```
so each course if the value is python truthy like a defined dict: it would pass.

So what's the purpose of creating another setting for courses, if the current could work in both versions

So in both examples I am checking the value of the key:
with the dict get
https://github.com/eduNEXT/eox-nelp/blob/7785b9bbc78aeabc5ffefd60ef62eb485dd91f77/eox_nelp/signals/receivers.py#L395-L397

Reply:
>So what's the purpose of creating another setting for courses, if the current could work in both versions

basically the name of the settings doesn't reflect the purpose of the setting

Reply:
I added in this 2 commits.
609622da0132b153cac73f2cb883aa330c8de03e
[f72c455](https://github.com/eduNEXT/eox-nelp/pull/216/commits/f72c4552434bdccaa863c8a802f8bf997dd17b6c)
Any way, I don't like it. Now, we have to manage two settings, one for v1 and the other for the engine.
Also, we have to synchronize those settings via query or something like that to configure each course with the Pearson version to use.

Reply:
1. You don't have to manage two settings previous implementation is deprecated  and you shouldn't use it
2. You didn't have to add the second condition that implementation fails if that setting is not present 
3. The previous implementation is deprecated therefore you shouldn't modify that at all
4. You don't need any query, it's faster if you do that manually there are only two courses